### ISSUE TITLE
[Error] AgoraRtmThread (31): EXC_BAD_ACCESS (code=1, address=0x18)

### ISSUE THREAD

**Issue by yukitaka13-1110** [AgoraIO/Flutter-RTM#26]
I use AgoraIO / Flutter-SDK and Flutter-RTM both.
I join RTC and RTM using same channel name.

When I call `dispose()`, this error occurred.

<img width="1184" alt="スクリーンショット 2020-02-25 19 48 46" src="https://user-images.githubusercontent.com/22259776/75241888-55c34800-580a-11ea-9952-b73ba84bd0ba.png">

I tried to catch exception but failed.
Even worse, any logs left.
This exception occurs sometimes.

What should I do ?

```dart
@override
  Future<void> dispose() async {
    super.dispose();
    try {
      await AgoraRtcEngine.leaveChannel();
      await AgoraRtcEngine.stopPreview();
      await AgoraRtcEngine.destroy();
      await _channelNameSubscription.cancel();
      if (_channel != null) {
        await _client?.releaseChannel(_channel.channelId);
      }
    } on Exception catch (e) {
      debugPrint(e.toString());
    }
  }
```

```
[✓] Flutter (Channel master, v1.15.4-pre.97, on Mac OS X 10.14.4 18E227, locale)
    • Flutter version 1.15.4-pre.97 at /xxx/flutter
    • Framework revision f81588a5a3 (6 days ago), 2020-02-18 16:33:55 -0800
    • Engine revision 77c5812d48
    • Dart version 2.8.0 (build 2.8.0-dev.9.0 999eeea5a3)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /xxxxx/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2.1, Build version 11B500
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.42.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

• No issued found!
```

**Comment by yukitaka13-1110** [comment]
I found this occurs when `client.logout()`
Not always. I can't understand how situation is triggered.

**Comment by LichKing-2234** [comment]
this is a bug of RTM native SDK, can you upgrade to the version `1.1.0` in your local `agora_rtm.podspec`

**Comment by yukitaka13-1110** [comment]
@LichKing-2234 Thank you for replying.
When will `agora_rtm` package be updated ?
Must I change `agora_rtm.podspec` version ?

**Comment by LichKing-2234** [comment]
Not so fast, pls change your local file before we upgrade

**Comment by yukitaka13-1110** [comment]
@LichKing-2234 
I couldn't update. Would you please teach me how to update?
```dart
 $ pod install
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "AgoraRtm_iOS":
  In snapshot (Podfile.lock):
    AgoraRtm_iOS (= 1.0.1, = 1.1.0)

  In Podfile:
    agora_rtm (from `.symlinks/plugins/agora_rtm/ios`) was resolved to 0.9.9, which depends on
      AgoraRtm_iOS (= 1.0.1)


You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * changed the constraints of dependency `AgoraRtm_iOS` inside your development pod `agora_rtm`.
   You should run `pod update AgoraRtm_iOS` to apply changes you've made.

```

**Comment by yukitaka13-1110** [comment]
@LichKing-2234 
I forked and push changes.
https://github.com/yukitaka13-1110/Flutter-RTM

For someone who encountering same problem.
When use this repository, specify like as follows.

```yaml
agora_rtm:
    git:
      url: git://github.com/yukitaka13-1110/Flutter-RTM.git
      ref: master
```
